tests/demos_test.go

```diff
@@ -0,0 +1,39 @@
+package tests_test
+
+import (
+	"io/ioutil"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	yaml "github.com/ghodss/yaml"
+
+	k8s "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Demo", func() {
+
+	Describe("Running reporing state", func() {
+		Context("periodically", func() {
+			AfterEach(func() {
+				dsClient := k8sClientset.AppsV1().DaemonSets("nmstate-default")
+				dsClient.Delete("state-controller", nil)
+			})
+			It("should report the correct state", func() {
+				manifest, err := ioutil.ReadFile("manifests/examples/state-controller-ds.yaml")
+				Expect(err).ShouldNot(HaveOccurred())
+				var ds k8s.DaemonSet
+				err = yaml.Unmarshal(manifest, &ds)
+				Expect(err).ShouldNot(HaveOccurred())
+				dsClient := k8sClientset.AppsV1().DaemonSets("nmstate-default")
+				_, err = dsClient.Create(&ds)
```

Comment:
why is ds created here, but removed in afterEach?

Reply:
I am reworking it to use beforeeach/aftereach to create/delete the ds